runtime/Cargo.toml

```
@@ -20,7 +20,7 @@ pallet-preimage = { branch = "moonbeam-polkadot-v0.9.19", default-features = fal
 pallet-proxy = { branch = "moonbeam-polkadot-v0.9.19", default-features = false, git = "https://github.com/purestake/substrate" }
 pallet-randomness-collective-flip = { branch = "moonbeam-polkadot-v0.9.19", default-features = false, git = "https://github.com/purestake/substrate" }
 pallet-scheduler = { branch = "moonbeam-polkadot-v0.9.19", default-features = false, git = "https://github.com/purestake/substrate" }
-pallet-sudo = { branch = "moonbeam-polkadot-v0.9.19", default-features = false, git = "https://github.com/purestake/substrate" }
+pallet-sudo = { branch = "moonbeam-polkadot-v0.9.19", default-features = false, optional = true, git = "https://github.com/purestake/substrate" }
```

Because of this, shall we add a `?` optional [here](https://github.com/zeitgeistpm/zeitgeist/blob/f3db56ff8945ffef19e696bc8d6c026b753c2a39/runtime/Cargo.toml#L306)? According to https://doc.rust-lang.org/cargo/reference/features.html#dependency-features

---

Good point, this was the initial idea why I made it optional. In the first iteration I used `pallet-sudo` and then `with-sudo` features. Turns out one cannot use the inherent `pallet-sudo` feature with the conditional inclusion `?`, and when using `with-sudo` one can use the `?` operator, but since `with-sudo` would be the default option, disabling it would require the rather unintuitive and too powerful flag `--no-default-features`. When using `without-sudo`, the `?` can't be used because `pallet-sudo` is used by default and only disabled with the feature.
So the way to go is to remove the optional parameter.

```suggestion
pallet-sudo = { branch = "moonbeam-polkadot-v0.9.19", default-features = false, git = "https://github.com/purestake/substrate" }
```

---

Thank you!